README.md
Templates for academic presentations -- pre-released
======================

> Beamer - LaTeX \ XeLaTeX templates for professional presentations

[![Build Status](https://api.travis-ci.org/demanasta/pres-templates.svg)](https://travis-ci.org/demanasta/pres-templates)
[![License MIT](http://img.shields.io/badge/license-MIT-brightgreen.svg)](LICENCE)
[![](https://img.shields.io/github/release/demanasta/pres-templates.svg)](https://github.com/demanasta/pres-templates/releases/latest)
[![](https://img.shields.io/github/tag/demanasta/pres-templates.svg)](https://github.com/demanasta/pres-templates/tags)

[![](https://img.shields.io/github/stars/demanasta/pres-templates.svg)](https://github.com/demanasta/pres-templates/stargazers)
[![](https://img.shields.io/github/forks/demanasta/pres-templates.svg)](https://github.com/demanasta/pres-templates/network)
[![](https://img.shields.io/github/issues/demanasta/pres-templates.svg)](https://github.com/demanasta/pres-templates/issues)


## Author(s)
*   Demitris G. Anastasiou	

--------------------------------------------------------------------------------
## Features

* Supports LaTeX, XeLaTeX and LuaLaTeX

* Support Greek Language and Fonts (GFS)

* Adaptive Title Page: Title page adapts to title length

* Pre-defined and custom fonts (Arial / Professional fonts) with math support

* Option to generate only specific chapters and references without the frontmatter and title page. Useful for review and corrections.

* Draft mode: Draft water mark, timestamp, version numbering
 
* Different styles and Classfiles:
    * `PhD`: A professional presentation for your thesis (PhD, Diploma etc)

    * `Pub`: Professional presentaions for your research work on conferences, meetings etc.
    
    * `Lct`: Academic presentation for university courses, lecture.

--------------------------------------------------------------------------------

## Building your thesis - XeLaTeX
### Using latexmk (Unix/Linux/Windows)

This template supports `XeLaTeX` compilation chain. To generate  PDF run

    latexmk -xelatex phd_pres.tex
    biber phd_pres.tex
    latexmk -xelatex -g phd_pres.tex
    
### Shell script for XeLaTeX (Unix/Linux) (Recommended)

Usage: `sh ./compile-thesis.sh [OPTIONS] [filename]`

[option]  xelatex: Compile the PhD thesis using xelatex

[option]  xelatexf: Compile xelatex and biber, full copilation

[option]  clean: removes temporary files no filename required

### Using the make file (Unix/Linux)  still some bugs

The template supports PDF, DVI and PS formats. All three formats can be generated
with the provided `Makefile`.

To build the `PDF` version of your thesis, run:

    make


This build procedure uses `latexmk` and will produce `phd_pres.pdf`.

Use `Variables.ini` to change options:

Clean unwanted files

To clean unwanted clutter (all LaTeX auto-generated files), run:

    make clean

__Bug__: You cannot use `printbib` option on Class file!! 

__Note__: the `Makefile` itself is take from and maintained at
[here](http://code.google.com/p/latex-makefile/).
 
-------------------------------------------------------------------------------

## Usage details

Thesis information such as title, author, year, degree, etc., and other meta-data can be modified in `pres-info.tex`

###Information file
All available informations can be modified in `pres-info.tex`

For each presentation (PhD, Pub, Lecture) there is a specific section in this file.

### Style file
All style files `.sty` placed at `sty` directory.
> Install Beamer custom style:
> 
> * Run: `$ tlmgr conf | grep "TEXMFHOME"
> 
> * Add style file at `${TEXMFHOME}/tex/latex/beamer/themes/`
> 
> * Run: `$ [sudo] texhash`

Otherwise you can place `.sty` file at the same directory with `.tex` file.

Then select you presentation style on `preample.tex` file
 
### Class files

* `beamerPhD`: Class file for thesis presentations

* `beamerPub`: Class file for conference presentations!

* `beamerLct`: Class file for conference presentation!

### Class options

* `aspectratio=169`: reduce ratio to 16:9

* `draft`: Special draft mode with line numbers, images, and water mark with timestamp and custom text. Position of the text can also be modified. To disable figures see on `preample.tex` the Draftmode section.

* `chapter`: This option enables only the specified chapter and it's references. Useful for review and corrections.

* `notes`: Prints frames and notes 

* `notes=only`: Prints only notes of each frame

* `printbib`: Include bibliography at the end of the presentation (__bug__: not working with makefile!)

* `progrbar`: Enable progress bar in the frame. Chose this option after your  first compilation.

## Contributing

1. Create an issue and describe your idea
2. [Fork it](https://github.com/demanasta/pres-templates/network#fork-destination-box)
3. Create your feature branch (`git checkout -b my-new-idea`)
4. Commit your changes (`git commit -am 'Add some feature'`)
5. Publish the branch (`git push origin my-new-idea`)
6. Create a new Pull Request
7. Profit! :white_check_mark:

## ChangeLog

The history of releases can be viewed at [ChangeLog](ChangeLog.md)

## Acknowlegments

* Xanthos Papanikolaou [@xanthospap](https://github.com/xanthospap) - Original design idea of presentation style 

ChangeLog.md
## Change log
> Not created yet!

### 2017/04/01 - Inception
> Author(s): Demitris G. Anastasiou


compile-pres.sh
#!/bin/bash
# A script to compile the PhD Thesis - Krishna Kumar 
# 24-01-2016: Updated to use xelatex Demitris G. Anastasiou
# 04-06-2017: Convert to compile presentations templates
# Distributed under GPLv2.0 License

#compile="compile";
xelatex="xelatex"; 
xelatexf="xelatexf";
clean="clean";

if test -z "$2"
then
if [ $1 = $clean ]; then
	echo "Cleaning please wait ..."
	rm -f *~
	rm -rf *.aux
	rm -rf *.bbl
	rm -rf *.blg
	rm -rf *.d
	rm -rf *.fls
	rm -rf *.ilg
	rm -rf *.ind
	rm -rf *.toc*
	rm -rf *.lot*
	rm -rf *.lof*
	rm -rf *.log
	rm -rf *.idx
	rm -rf *.out*
	rm -rf *.nlo
	rm -rf *.nls
	rm -rf *.bcf
	rm -rf *.bib
	rm -rf *.fdb_latexmk
	rm -rf *.nav
	rm -rf *.run.xml
	rm -rf *.snm
	rm -rf $filename.pdf
	rm -rf $filename.ps
	rm -rf $filename.dvi
	rm -rf *#* 
	echo "Cleaning complete!"
	exit
else
	echo "Shell script for compiling the PhD Thesis"
	echo "Usage: sh ./compile-thesis.sh [OPTIONS] [filename]"
#	echo "[option]  compile: Compiles the PhD Thesis"
	echo "[option]  xelatex: Compile the PhD thesis using xelatex"
	echo "[option]  xelatexf: Compile xelatex and biber, full copilation"
	echo "[option]  clean: removes temporary files no filename required"
	exit
fi
fi

filename=$2;

if [ $1 = $clean ]; then
	echo "Cleaning please wait ..."
        rm -f *~
        rm -rf *.aux
        rm -rf *.bbl
        rm -rf *.blg
        rm -rf *.d
        rm -rf *.fls
        rm -rf *.ilg
        rm -rf *.ind
        rm -rf *.toc*
        rm -rf *.lot*
        rm -rf *.lof*
        rm -rf *.log
        rm -rf *.idx
        rm -rf *.out*
        rm -rf *.nlo
        rm -rf *.nls
        rm -rf *.bcf
        rm -rf *.bib
        rm -rf *.fdb_latexmk
        rm -rf *.nav
        rm -rf *.run.xml
        rm -rf *.snm
        rm -rf $filename.pdf
        rm -rf $filename.ps
        rm -rf $filename.dvi
        rm -rf *#* 
        echo "Cleaning complete!"
        exit
elif [ $1 = $xelatexf ]; then
	echo "Compiling your PhD Thesis...please wait...!"
	latexmk -xelatex $filename.tex				# <<-- NEW compilation after 26 May 17
	biber $filname.aux
	latexmk -xelatex -g $filename.tex
	echo "Success!"
	echo "LAST RUN" $(date)
	exit
elif [ $1 = $xelatex ]; then
        echo "Compiling your PhD Thesis...please wait...!"
#        latexmk -xelatex $filename.tex                          # <<-- NEW compilation after 26 May 17
#        biber $filname.aux
#        latexmk -xelatex -g $filename.tex
        xelatex $filename.tex
        echo "Success!"
	echo "LAST RUN:" $(date)
        exit
else
	echo "Shell script for compiling the PhD Thesis"
	echo "Usage: sh ./compile-thesis.sh [OPTIONS] [filename]"
#	echo "[option]  compile: Compiles the PhD Thesis"
	echo "[option]  xelatex: Compile the PhD thesis using xelatex"
        echo "[option]  xelatexf: Compile xelatex and biber, full copilation"
	echo "[option]  clean: removes temporary files no filename required"
	exit
fi

# reduce size of pdf file... sometimes work
# gs -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/default     -dNOPAUSE -dQUIET -dBATCH -dDetectDuplicateImages     -dCompressFonts=true -r80 -sOutputFile=phd_pres_low.pdf phd_pres.pdf

if test -z "$3"
then
	exit
fi


compile-phd_pres.sh
#!/bin/bash
# A script to compile the PhD Thesis - Krishna Kumar 
# 24-01-2016: Updated to use xelatex Demitris G. Anastasiou
# 04-06-2017: Convert to compile presentations templates
# Distributed under GPLv2.0 License

#compile="compile";
xelatex="xelatex"; 
xelatexf="xelatexf";
clean="clean";

if test -z "$2"
then
if [ $1 = $clean ]; then
	echo "Cleaning please wait ..."
	rm -f *~
	rm -rf *.aux
	rm -rf *.bbl
	rm -rf *.blg
	rm -rf *.d
	rm -rf *.fls
	rm -rf *.ilg
	rm -rf *.ind
	rm -rf *.toc*
	rm -rf *.lot*
	rm -rf *.lof*
	rm -rf *.log
	rm -rf *.idx
	rm -rf *.out*
	rm -rf *.nlo
	rm -rf *.nls
	rm -rf *.bcf
	rm -rf *.bib
	rm -rf *.fdb_latexmk
	rm -rf *.nav
	rm -rf *.run.xml
	rm -rf *.snm
	rm -rf $filename.pdf
	rm -rf $filename.ps
	rm -rf $filename.dvi
	rm -rf *#* 
	echo "Cleaning complete!"
	exit
else
	echo "Shell script for compiling the PhD Thesis"
	echo "Usage: sh ./compile-thesis.sh [OPTIONS] [filename]"
#	echo "[option]  compile: Compiles the PhD Thesis"
	echo "[option]  xelatex: Compile the PhD thesis using xelatex"
	echo "[option]  xelatexf: Compile xelatex and biber, full copilation"
	echo "[option]  clean: removes temporary files no filename required"
	exit
fi
fi

filename=$2;

if [ $1 = $clean ]; then
	echo "Cleaning please wait ..."
        rm -f *~
        rm -rf *.aux
        rm -rf *.bbl
        rm -rf *.blg
        rm -rf *.d
        rm -rf *.fls
        rm -rf *.ilg
        rm -rf *.ind
        rm -rf *.toc*
        rm -rf *.lot*
        rm -rf *.lof*
        rm -rf *.log
        rm -rf *.idx
        rm -rf *.out*
        rm -rf *.nlo
        rm -rf *.nls
        rm -rf *.bcf
        rm -rf *.bib
        rm -rf *.fdb_latexmk
        rm -rf *.nav
        rm -rf *.run.xml
        rm -rf *.snm
        rm -rf $filename.pdf
        rm -rf $filename.ps
        rm -rf $filename.dvi
        rm -rf *#* 
        echo "Cleaning complete!"
        exit
#elif [ $1 = $compile ]; then
#	echo "Compiling your PhD Thesis...please wait...!"
#	pdflatex -interaction=nonstopmode $filename.tex
#	bibtex $filename.aux 	
#	makeindex $filename.aux
#	makeindex $filename.idx
#	makeindex $filename.nlo -s nomencl.ist -o $filename.nls
#	pdflatex -interaction=nonstopmode $filename.tex
#	makeindex $filename.nlo -s nomencl.ist -o $filename.nls
#	pdflatex -interaction=nonstopmode $filename.tex
#	echo "Success!"
#	exit
elif [ $1 = $xelatexf ]; then
	echo "Compiling your PhD Thesis...please wait...!"
	latexmk -xelatex $filename.tex				# <<-- NEW compilation after 26 May 17
	biber $filname.aux
	latexmk -xelatex -g $filename.tex
	echo "Success!"
	echo "LAST RUN" $(date)
	exit
elif [ $1 = $xelatex ]; then
        echo "Compiling your PhD Thesis...please wait...!"
#        latexmk -xelatex $filename.tex                          # <<-- NEW compilation after 26 May 17
#        biber $filname.aux
#        latexmk -xelatex -g $filename.tex
        xelatex $filename.tex
        echo "Success!"
	echo "LAST RUN:" $(date)
        exit
else
	echo "Shell script for compiling the PhD Thesis"
	echo "Usage: sh ./compile-thesis.sh [OPTIONS] [filename]"
#	echo "[option]  compile: Compiles the PhD Thesis"
	echo "[option]  xelatex: Compile the PhD thesis using xelatex"
        echo "[option]  xelatexf: Compile xelatex and biber, full copilation"
	echo "[option]  clean: removes temporary files no filename required"
	exit
fi

# reduce size of pdf file... sometimes work
# gs -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/default     -dNOPAUSE -dQUIET -dBATCH -dDetectDuplicateImages     -dCompressFonts=true -r80 -sOutputFile=phd_pres_low.pdf phd_pres.pdf

if test -z "$3"
then
	exit
fi
